src/Modules/OrchardCore.Commerce/Services/ProductService.cs

```diff
@@ -76,4 +76,18 @@ private static void FillField(ContentPart part, ContentPartFieldDefinition field
             part.Get(typeOfField, fieldName);
         }
     }
+
+    public string GetVariantKey(string sku)
```

Comment:
Sorry, I just noticed something. `productPart.Sku + '-' + value.Classify().ToUpperInvariant` This part is missing.

> Oliver is right, you only save the base SKU. The variant SKU should be generated similarly to [how the price variant labels are generated](https://github.com/OrchardCMS/OrchardCore.Commerce/blob/main/src/Modules/OrchardCore.Commerce/Drivers/PriceVariantsPartDisplayDriver.cs#L82) and then converted to that upper case format I previously described.